### PR TITLE
Test example bash snippets

### PR DESCRIPTION
**Pull request by Twey**
## Motivation

We would like to ensure that bash snippets in our example READMEs are not broken.

## Proposal

Add tests that execute the bash in the examples.

## Test Plan

Run `cargo test -p linera-service`.

## Release Plan

These changes only affect testing and should be safe to release.

## Links

- [reviewer checklist](https://github.com/linera-io/linera-protocol/blob/main/CONTRIBUTING.md#reviewer-checklist)
- https://github.com/linera-io/linera-protocol/issues/1180
- https://github.com/linera-io/linera-protocol/issues/1176
- https://github.com/linera-io/linera-protocol/issues/1177

### CODE REVIEW

**Comment by ma2bd** [comment]
Need to update Cargo.lock

**Comment by ma2bd** [comment]
> Extract the owner and chain IDs programmatically rather than hard-coding them into the script and relying on a deterministic RNG seed.

FYI chain IDs are actually always deterministic (based on block heights, transaction index, etc)

**Comment by Twey** [comment]
> Need to update Cargo.lock

Yes sorry, I pushed the commit for debugging purposes last night and then went to bed while the CI did its thing :)  I was trying to keep the `Cargo.lock` as similar to `main` as possible.

This commit is just for debugging purposes (generally commits labelled `XXX` are intended to not make it into the final merge) but this response from the CI is quite interesting:

Thinking that perhaps I'd accidentally pulled in a different version of `syn` when updating `test-case` I added a debugging commit that reverts that bump but does nothing else — so all `Cargo.toml` and `Cargo.lock` files should be in exactly the same state as on `main` with the singular exception that `test-case`.  But in fact on some inspection of the `syn` source code the symbols that `check-all-features` is complaining about have always been gated behind `full` back to their introduction, so it can't be a version issue.

The other errors around unused imports seem to be symptomatic of this error.

> FYI chain IDs are actually always deterministic (based on block heights, transaction index, etc)

On some consideration I think this is an invalid complaint anyway — insofar as the documentation is supposed to reflect a reasonable ‘interactive’ session, the user would probably just copy/paste the IDs out of the `linera wallet show` output.  I've removed it :)

**Comment by Twey** [comment]
I've pushed a commit that fixes the CI — but I'm not sure that it really belongs in this pull request.  What do we think?

**Comment by ma2bd** [comment]
> I've pushed a commit that fixes the CI — but I'm not sure that it really belongs in this pull request. What do we think?

It must somehow belong here because CI is passing on the `main` branch.

File: Cargo.lock
Comment (ma2bd) [inline Cargo.lock:726]: It looks like you run `cargo update`?

**Review by ma2bd — APPROVED** [review]
LGTM

**Comment by ma2bd** [comment]
> The two application publishes are racy — ..
> The two node service spawns are racy — ..

I think I would add a small `sleep` in the bash script to hint that something is going on. And/or add a TODO with a reference to a github issue.

File: examples/social/src/lib.rs
Comment (Twey) [reply examples/social/src/lib.rs:4]: I tend to use block comments for large blocks of documentation since it makes it easier to edit — if it's against Linera policy let me know and I can switch it back :)

File: Cargo.lock
Comment (Twey) [reply Cargo.lock:726]: For the fix commit yes, and also probably in the build after updating `test-case`

File: examples/crowd-funding/README.md
Comment (Twey) [reply examples/crowd-funding/README.md:103]: I've refilled at 90.  Additionally I've added a `.editorconfig` file with this and other standards you've mentioned in PR https://github.com/linera-io/linera-protocol/pull/1175.

**Comment by Twey** [comment]
>I think I would still add a small sleep in the bash script to hint that something is going on, and ideally add a TODO with a reference to a github issue.

I've created https://github.com/linera-io/linera-protocol/issues/1176 and https://github.com/linera-io/linera-protocol/issues/1177 and referenced them from comments around the documentation.

**Comment by Twey** [comment]
I'm waiting for https://github.com/linera-io/linera-protocol/pull/1178 to merge this since once that's merged I can take out the two dependency-change PRs.

File: Cargo.lock
Comment (Twey) [reply Cargo.lock]: For the final version I've removed the `test-case` update from here, preferring to move it to https://github.com/linera-io/linera-protocol/pull/1178 with others of its kind.  In order to add `test-case` to `linera-service` without doing a full `cargo update`, I manually edited `Cargo.lock` to add only the dependency we need to build without updating `Cargo.lock`.

**Pull request merged**
